Sources/SwiftParser/Patterns.swift

```diff
@@ -91,32 +86,48 @@ extension Parser {
           arena: self.arena
         )
       )
-    case (.let, let handle)?,
-      (.var, let handle)?,
-      (.inout, let handle)?:
-      let bindingSpecifier = self.eat(handle)
-      let value = self.parsePattern()
-      return RawPatternSyntax(
-        RawValueBindingPatternSyntax(
-          bindingSpecifier: bindingSpecifier,
-          pattern: value,
-          arena: self.arena
-        )
-      )
-    case nil:
-      if self.currentToken.isLexerClassifiedKeyword, !self.atStartOfLine {
-        // Recover if a keyword was used instead of an identifier
-        let keyword = self.consumeAnyToken()
+    case (.rhs(let binding), let handle)?:
+      switch binding {
+      case ._mutating:
+        fallthrough
+      case ._borrowing:
+        fallthrough
+      case ._consuming:
+        guard experimentalFeatures.contains(.referenceBindings) else {
+          break
+        }
+        fallthrough
+      case .let:
+        fallthrough
+      case .var:
+        fallthrough
+      case .inout:
+        // Handle all cases of `ValueBindingPatternSyntax.BindingSpecifierOptions` in the same way.
```

Comment:
I don’t think we need this comment anymore if we exhaustively switch over `rhs`.